### ISSUE TITLE
Image occlusion and always different questionnaire

### ISSUE THREAD

**Issue by LeonelRFF** [ECuiDev/obsidian-quiz-generator#18]
Hello first of all great plugin, I see that it will incorporate questions based on images and pdf in future versions, so I share my ideas

1.  The possibility of incorporating image occlusion type cards

Use multimodal models to recognize text in images to cover said text automatically, I would like this to be used with the excalidraw plugin so that it is compatible, here is an example of how it can be made simple

https://github.com/user-attachments/assets/2b86a7d7-6d44-40fa-aa7b-21dbe4886c1c

2.  Always different questions

this is related to **testo mode** and **Question Variety**, the idea is that sometimes there are concepts that must be memorized, there is no other way, but on other occasions they must only be understood in their different situations, the clearest example of this is mathematics, providing the LLM as a source the notes or PDFs (and other sources if necessary) so that in each new study session, the exercises are always new and based on this evaluation _**the next session will be more difficult**_ 🫡

**Comment by ECuiDev** [comment]
For your second idea, do you mean the generated questions change depending on what questions were generated previously, and the difficulty changes depending on how many questions you answered correctly?

**Comment by LeonelRFF** [comment]
Yes, I thought about it like this, if a question that was created was later evaluated and answered well, then in the next session that same question will increase in difficulty. Here I show you a diagram, I hope I express myself well, I am not a programmer.

```mermaid
graph TD
    A[Create Question] --> B[Deck]
    B --> C[Evaluate Question]
    C --> D{How did you respond?}
    D --> E[Bad or so-so]
    E --> F[Remains the same]
    F --> C
    D --> G[Good]
    G --> H{How many times did you answer correctly?}
    H --> I[Only once]
    H --> J[More than once]
    I --> F
    J --> K[Increase difficulty of this card only]
    K --> A

**Comment by ECuiDev** [comment]
Oh, I see. I think I have a pretty good idea of what you're talking about (the diagram is very helpful). This is a really interesting idea so I'll likely include it as part of the **Test Mode** feature.

**Comment by LeonelRFF** [comment]
Great, to clarify further I see that I forgot to add after more than once that said card that has already been answered correctly **more than once** will return to the deck but in "learned" mode so that it is not shown again, but also not deleted, I will only be able to see it if I unlock it for the final review 😉
